src/core/Val.ml

```diff
@@ -21,7 +21,12 @@ type error =
   | RigidCoeUnexpectedArgument of abs
   | RigidHComUnexpectedArgument of value
   | RigidGHComUnexpectedArgument of value
+  | ApplyUnexpectedFunction of value
+  | ApplyUnexpectedCube of value
```

Comment:
Not sure if `ApplyUnexpectedCube` is the best name....

Reply:
It is not too bad...